crates/wasmtime/src/trampoline/func.rs

```diff
@@ -43,6 +43,11 @@ unsafe extern "C" fn stub_fn<F>(
     // have any. To prevent leaks we avoid having any local destructors by
     // avoiding local variables.
     let result = panic::catch_unwind(AssertUnwindSafe(|| {
+        // The general type for the first context argument is `*mut ()` but
+        // within the context of this function we know that this is `VMContext`
+        // due to the compilation of this function. Note that this is
+        // debug-asserted in `host_state()` with the `VMCONTEXT_MAGIC` value.
+        let vmctx = vmctx as *mut VMContext;
```

Comment:
Can we make the `*mut ()` to `*mut VMContext` cast (that we've done in two different places now) go through a helper function that does the magic value debug assertion eagerly?